src/python/grpcio_tests/tests/fork/methods.py

```diff
@@ -118,27 +124,114 @@ def __init__(self, task, args=None):
         if args is None:
             args = ()
         self._exceptions = multiprocessing.Queue()
+        self._stdout_path = tempfile.mkstemp()[1]
+        self._stderr_path = tempfile.mkstemp()[1]
+        self._child_pid = None
+        self._rc = None
+        self._args = args
+
+        self._task = task
 
-        def record_exceptions():
-            try:
-                task(*args)
-            except grpc.RpcError as rpc_error:
-                self._exceptions.put('RpcError: %s' % rpc_error)
-            except Exception as e:  # pylint: disable=broad-except
-                self._exceptions.put(e)
+    def _child_main(self):
+        import faulthandler
+        faulthandler.enable(all_threads=True)
 
-        self._process = multiprocessing.Process(target=record_exceptions)
+        try:
+            self._task(*self._args)
+        except grpc.RpcError as rpc_error:
+            traceback.print_exc()
+            self._exceptions.put('RpcError: %s' % rpc_error)
+        except Exception as e:  # pylint: disable=broad-except
+            traceback.print_exc()
+            self._exceptions.put(e)
+        sys.exit(0)
+
+    def _orchestrate_child_gdb(self):
+        cmd = [
+            "gdb",
```

Comment:
Note to self: Somehow ensure GDB is installed in the test execution environment.

Reply:
It turns out this won't work at the moment given that we need to change a kernel setting. We can leave it for local development though. For reference, you need to run `sudo bash -c 'echo "0" > /proc/sys/kernel/yama/ptrace_scope'` to get this working locally.